src/interactive-window/editor-integration/cellHashProviderFactory.ts

```diff
@@ -43,13 +45,18 @@ export class CellHashProviderFactory {
         if (existing) {
             return existing;
         }
+        const nb = getAssociatedNotebookDocument(kernel);
+        if (!nb) {
+            throw new Error('No notebook associated with the kernel');
```

Comment:
This should never happen, we ensure we never call this code for kernels created by anything other than Jupyter extesnsion.